internal/app/networkd/pkg/nic/nic.go

```diff
@@ -187,11 +188,7 @@ func (n *NetworkInterface) Addressing() error {
 	for _, method := range n.AddressMethod {
 		if err := n.configureInterface(method); err != nil {
 			// Treat as non fatal error when failing to configure an interface
-			return nil
-		}
-
-		if !method.Valid() {
-			return nil
```

Comment:
continue instead of return early. This function could probably use a little bit of a refactor ( not return a value since we don't ever return an error ), but that can wait till later.